Code/Tools/TestImpactFramework/Frontend/Console/Common/Code/Include/Static/TestImpactTestSequenceNotificationHandler.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <TestImpactFramework/TestImpactTestSequenceBus.h>
+
+#include <AzCore/std/chrono/chrono.h>
+#include <AzCore/std/containers/unordered_set.h>
+#include <AzCore/std/string/string.h>
+
+#pragma once
+
+namespace TestImpact
+{
+    namespace Console
+    {
+        //! Console output dumping for test target standard output and error.
+        enum class ConsoleOutputMode : AZ::u8
+        {
+            Buffered, //!< Output will be dumped to console once the test target execution has completed.
+            Realtime //!< Output will be dumped to console in realtime.
+        };
+
+        //! Base class for all test sequence notification handlers.
+        class TestSequenceNotificationHandlerBase
+            : private TestSequenceNotificationsBaseBus::Handler
+        {
+        public:
+            TestSequenceNotificationHandlerBase(ConsoleOutputMode consoleOutputMode);
+            ~TestSequenceNotificationHandlerBase();
```

Comment:
Need to mark the destructors of these classes as virtual? Or at least the base one?

Reply:
That or make them protected and non-virtual (do protect against trying to delete the base pointer)

Reply:
Fixed 👍